src/Core/src/Graphics/ShapeDrawable.cs

```diff
@@ -35,24 +38,68 @@ public void UpdateRenderTransform(Matrix3x2? renderTransform)
 
 		public void Draw(ICanvas canvas, RectF dirtyRect)
 		{
-			var rect = dirtyRect;
+			if (ShapeView == null)
+				return;
 
 			IShape? shape = ShapeView?.Shape;
 
 			if (shape == null)
 				return;
 
+			var rect = dirtyRect;
+			
 			PathF? path = shape.PathForBounds(rect);
 
 			if (path == null)
 				return;
 
+			AdjustShape(canvas, dirtyRect, path);
 			ApplyTransform(path);
 
 			DrawStrokePath(canvas, rect, path);
 			DrawFillPath(canvas, rect, path);
 		}
 
+		void AdjustShape(ICanvas canvas, RectF dirtyRect, PathF path)
+		{
+			// Adjust the Shape position and size based on the
```

Comment:
Here is where we position and adjust the correct size based on the StrokeThickness property.